src/v/utils/s3_fifo.h

```diff
@@ -314,10 +313,12 @@ class cache_hook {
 
     friend class testing_details::cache_hook_accessor;
 
+    hook_type hook_;
+    // This is conceptually an optional, but that increases the struct size.
+    uint64_t ghost_insertion_time_ = 0;
+    bool has_ghost_insertion_time_ = false;
```

Comment:
nit: maybe we want to use uniform initialization here to match `freq_` and `location_`.

Reply:
Done in latest force push